### ISSUE TITLE
Add support for SSL connections to gearmand

### ISSUE THREAD

**Issue by esabol** [php/pecl-networking-gearman#9]
See https://github.com/wcgallego/pecl-gearman/pull/72 for a patch. Any thoughts?

**Comment by esabol** [comment]
Hello? Is there a maintainer around?

**Comment by esabol** [comment]
@oleg-st, any thoughts on this?

**Comment by oleg-st** [comment]
@esabol I do not mind, but I am not a maintainer :)

**Comment by esabol** [comment]
> @esabol I do not mind, but I am not a maintainer :)

Ah, I thought you were. Sorry, I misread the commits.

**Comment by esabol** [comment]
Is anyone interested in this feature? I'd love to collaborate with someone on incorporating my very outdated branch of the old repository into this repository.

**Comment by esabol** [comment]
Well, I think I've finally done it. Before I submit a PR, could someone look over my changes on adding support for the `setSSL()` method?

https://github.com/esabol/pecl-networking-gearman/commit/b1abea91c31a7d238faa300d172f024ba9e56344

It works perfectly in my testing using my gearmand server with SSL turned on.

I've added some tests to the test suite, but they only test the method with setSSL set to false. I think it would be rather difficult to test otherwise in most scenarios. Not all libgearman.so's are compiled with SSL support for one.

**Comment by esabol** [comment]
I just noticed that `gearman_arginfo.h` has a comment that says:
```
/* This is a generated file, edit the .stub.php file instead.
```
Well, I already edited `gearman_arginfo.h`. Oops. How do I properly generate the `gearman_arginfo.h` from the `gearman.stub.php` file?

**Comment by oleg-st** [comment]
@esabol Use `make` or `make gearman_arginfo.h` or `php build/gen_stub.php gearman.stub.php`

**Comment by esabol** [comment]
@oleg-st wrote:
> Use `make` or `make gearman_arginfo.h` or `php build/gen_stub.php gearman.stub.php`

Thanks, Oleg! I went with `make gearman_arginfo.h`. The generated `gearman_arginfo.h` has a lot of differences compared to the version in the repo though. That would seem to indicate errors in the `gearman.stub.php` file, no? What am I supposed to do about those with regards to the PR I'm working on? It sure seems like people have been editing `gearman_arginfo.h` manually.

**Comment by esabol** [comment]
> I think `gearman.stub.php` was changed here #24

Yeah, and that PR neglected to update `gearman_arginfo.h`, so here we are, I guess.

`make gearman_arginfo.h` fails on PHP 8.4. It complains about the following:
```
In gearman.stub.php:
GearmanClient::addServer(): Parameter host has null default, but is not nullable
```
